components/base/kinematicbase/execution.go

```diff
@@ -513,9 +524,23 @@ func (ptgk *ptgBaseKinematics) makeCourseCorrectionGoals(
 	for i := currStep; i < len(steps); i++ {
 		for len(steps[i].subTraj)-startingTrajPt > stepsRemainingThisGoal {
 			goalTrajPtIdx := startingTrajPt + stepsRemainingThisGoal
+
+			// Since the arc may not be starting at 0, we must compute the transform for this particular traj pt.
+			// The pose in trajPt.Pose is from the zero position.
+			arcTrajInputs := []referenceframe.Input{
+				steps[i].arcSegment.StartConfiguration[ptgIndex],
+				steps[i].arcSegment.StartConfiguration[trajectoryAlphaWithinPTG],
+				steps[i].arcSegment.StartConfiguration[startDistanceAlongTrajectoryIndex],
+				{steps[i].subTraj[goalTrajPtIdx].Dist},
+			}
+			arcPose, err := ptgk.Kinematics().Transform(arcTrajInputs)
+			if err != nil {
+				return []courseCorrectionGoal{}
+			}
+
 			goalPose := spatialmath.PoseBetween(
 				currPose,
-				spatialmath.Compose(steps[i].arcSegment.StartPosition, steps[i].subTraj[goalTrajPtIdx].Pose),
+				spatialmath.Compose(steps[i].arcSegment.StartPosition, arcPose),
```

Comment:
Noticing a proliferation of code that composes these two Poses. Am I right in assuming they might simplify with @nfranczak 's work on "execution frames"

Reply:
Probably not super related to the execution frame work.

We're requesting a relative transform along an arc (arcPose) and then determining where that pose will place us in absolute terms relative to other poses we care about (arcSegment.StartPosition). Whether the absolute pose onto which we are composing comes from a segment, an execution frame, or something else, will all have this same paradigm.